packages/internal/src/generate/typeDefinitions.ts

```diff
@@ -219,57 +215,75 @@ export const generateTypeDefGraphQLApi = async () => {
 }
 
 export const generateTypeDefGraphQLWeb = async () => {
-  if (findCells().length) {
-    const rwjsPaths = getPaths()
-    try {
-      const f = await generateTypeDefGraphQL({
-        [path.join(rwjsPaths.web.types, 'graphql.d.ts')]: {
-          documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
-          plugins: [
-            {
-              typescript: {
-                enumsAsTypes: true,
-              },
+  const rwjsPaths = getPaths()
+  try {
+    const f = await runCodegenGraphQL({
+      [path.join(rwjsPaths.web.types, 'graphql.d.ts')]: {
+        documents: './web/src/**/!(*.d).{ts,tsx,js,jsx}',
+        plugins: [
+          {
+            typescript: {
+              enumsAsTypes: true,
             },
-            'typescript-operations',
-          ],
-        },
-      })
-      return f
-    } catch (e) {
-      console.error()
-      console.error('Error: Could not generate GraphQL type definitions (web)')
-      console.error()
-      return []
-    }
-  } else {
+          },
+          'typescript-operations',
+        ],
+      },
+    })
+    return f
+  } catch (e) {
+    console.error()
+    console.error('Error: Could not generate GraphQL type definitions (web)')
+    console.error()
     return []
   }
 }
 
-const generateTypeDefGraphQL = async (generates: Record<string, unknown>) => {
+/**
+ * This is the function used internally by generateTypeDefGraphQLApi and generateTypeDefGraphQLWeb
+ * And contains the base configuration for generating gql types with codegen
+ *
+ * Named a little differently to make it easier to spot
+ */
+
+// CLI generate also takes cwd
+type CodegenConfig = CodegenTypes.Config & { cwd: string }
+
+const runCodegenGraphQL = async (
+  generates: Record<string, CodegenTypes.ConfiguredOutput>
+) => {
   const rwjsPaths = getPaths()
   type GenerateResponse = { filename: string; contents: string }[]
-  // https://www.graphql-code-generator.com/docs/getting-started/programmatic-usage#using-the-cli-instead-of-core
-  const f: GenerateResponse = await generate(
-    {
-      cwd: rwjsPaths.base,
-      schema: rwjsPaths.generated.schema,
-      config: {
-        scalars: {
-          DateTime: 'string',
-          Date: 'string',
-          JSON: 'Record<string, unknown>',
-          JSONObject: 'Record<string, unknown>',
-          Time: 'string',
-        },
-        omitOperationSuffix: true, // prevent type names being PetQueryQuery, RW generators already append Query/Mutation/etc.
+
+  const codegenConfig: CodegenConfig = {
+    cwd: rwjsPaths.base,
+    schema: rwjsPaths.generated.schema,
+    config: {
+      namingConvention: 'keep', // to allow camelCased query names
```

Comment:
👍 Just want to make sure we're okay with camel cases **types** too. What do you think? No biggie IMO. 😁

Reply:
I looked into it, and there's no reason the query names have to be PascalCase, so camcelCase or UPPERCASE or whatever they choose should be fine

Reply:
Right right. I was thinking more of the generated type (not the query operation name itself).